camera/basic/src/main/java/com/sample/camera/basic/CameraActivity.java

```diff
@@ -212,19 +214,53 @@ public void RequestCamera() {
     public void onRequestPermissionsResult(int requestCode,
                                            @NonNull String[] permissions,
                                            @NonNull int[] grantResults) {
-        /*
-         * if any permission failed, the sample could not play
-         */
-        if (PERMISSION_REQUEST_CODE_CAMERA != requestCode) {
+        if (requestCode != PERMISSION_REQUEST_CODE_CAMERA) {
+            // The permissions request isn't ours.
             super.onRequestPermissionsResult(requestCode,
                                              permissions,
                                              grantResults);
             return;
         }
 
-        if(grantResults.length == 2) {
-            notifyCameraPermission(grantResults[0] == PackageManager.PERMISSION_GRANTED &&
-                    grantResults[1] == PackageManager.PERMISSION_GRANTED);
+        if (permissions.length == 0) {
+            // https://developer.android.com/reference/androidx/core/app/ActivityCompat.OnRequestPermissionsResultCallback#onRequestPermissionsResult(int,java.lang.String[],int[])
+            //
+            // Note: It is possible that the permissions request interaction with the user is
+            // interrupted. In this case you will receive empty permissions and results arrays which
+            // should be treated as a cancellation.
+            //
+            // The docs aren't clear about *why* is might be canceled, so it's not clear what we
```

Comment:
s/is/it/

Reply:
can you find any 1p app code and see what they're doing? (and/or file a bug against the docs.)

Reply:
Yeah, hadn't quite gotten to filing a bug yet but that was the plan. http://b/340568363